### ISSUE TITLE
CA1508 False Positive in nested catch block

### ISSUE THREAD

**Issue by chrissmiller** [dotnet/roslyn-analyzers#5968]
### Analyzer

**Diagnostic ID**: [CA1508](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/CA1508): `CA1508: Avoid dead conditional code`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://docs.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 5.0.406](https://dotnet.microsoft.com/download/dotnet/5.0)

### Describe the bug
When catching an exception inside a catch block for another exception of the same type, CA1508 identifies any conditionals on the second catch which conflict with conditionals on the first catch as being dead code

### Steps To Reproduce
```
try
{
    throw Exception("A");
}
catch (Exception ex) when (ex.Message == "A")
{
    try
    {
        throw Exception("B");
    }
    // CA1508: 'ex2.Message == "B"' is always 'false'. Remove or refactor the condition(s) to remove dead code.
    catch (Exception ex2) when (ex2.Message == "B")
    {
        // do something
    }
}
```

**Comment by mavasani** [comment]
Verified this repros on .NET 7 SDK, but not on the latest .NET 8 preview SDK.
